docs/references/strategies/languages/golang/gomodules.md

```diff
@@ -235,3 +235,24 @@ not using backoff!
 
 As a concrete step towards resolving this sort of discrepancy, we recommend running `go mod tidy` on projects regularly;
 this command should synchronize the `go.mod` file with the actual state of the project.
+
+#### Test Dependencies
+
+Sometimes the above procedure may uncover a dependency that is not reported by FOSSA but that is also not removed by `go mod tidy`.
+The other reason that a dependency may appear in `go.mod` but not in FOSSA is that it is a test dependency.
+`go.mod` itself does not label dependencies as being used only in tests, but FOSSA's Go module strategy can identify these and will exclude them.
+To verify that a direct dependency is not a test-only dependency, we recommend searching the project source code for where a package from the module is imported.
+Generally, for a declaration in `go.mod` like:
+
+```
+	github.com/prometheus/client_golang v1.12.2
+```
+
+You can search for its imports using a command like:
+
+```sh
+$ find <project_directory> -name \*.go -exec grep -Hn "github.com/prometheus/client_golang" {} \;
+```
+
+Then verify that it appears in at least one Go source file _not_ ending in `_test.go`.
```

Comment:
🤔 what if their non-`_test.go` file is only called by a `_test.go` file? what do we report & what does Go consider it as in that scenario?

Reply:
I made a quick example project and we report that the dep in the non-test file is a regular dependency and not just a test dep, even though in practice it is only used in a test-suite. I think that's because the non`_test.go` file is still part of a larger package and so could theoretically be called from non-test code. Strangely, this did not change even when I made the function called from the test code private to the package.

I think this means that the find/grep method is generally consistent with what we report and what Go considers packages to be, but please ask questions if the above doesn't make much sense.

Reply:
Yeah, that tracks. I think Go's code pruning isn't granular enough to notice this and this is roughly what I thought would be the case. Thanks for checking!